shipyard_blueprints/tableau/shipyard_tableau/tableau_utils.py

```diff
@@ -0,0 +1,217 @@
+import tableauserverclient as TSC
+from shipyard_templates import ShipyardLogger, ExitCodeException, DataVisualization
+
+EXIT_CODE_FILE_WRITE_ERROR = 100
+
+EXIT_CODE_INVALID_PROJECT = 201
+EXIT_CODE_INVALID_WORKBOOK = 202
+EXIT_CODE_INVALID_VIEW = 203
+EXIT_CODE_INVALID_JOB = 204
+EXIT_CODE_INVALID_DATASOURCE = 205
+EXIT_CODE_REFRESH_ERROR = 206
+
+EXIT_CODE_FINAL_STATUS_CANCELLED = 210
+EXIT_CODE_FINAL_STATUS_ERRORED = 211
+EXIT_CODE_STATUS_INCOMPLETE = 212
+
+logger = ShipyardLogger.get_logger()
+
+
+def connect_to_tableau(
+    username: str, password: str, site_id: str, server_url: str, sign_in_method: str
+):
+    """
+    Helper function to establish connection and server
+
+    Args:
+        username (str): Tableau username
+        password (str): Tableau password
+        site_id (str): Tableau site_id
+        server_url (str): Tableau server_url
+        sign_in_method (str): Tableau sign_in_method
+    Returns:
+        server: Tableau server
+        connection: Tableau connection
+    """
+    try:
+        if site_id.lower() == "default":
+            site_id = ""
+
+        if sign_in_method == "access_token":
+            tableau_auth = TSC.PersonalAccessTokenAuth(
+                token_name=username, personal_access_token=password, site_id=site_id
+            )
+
+        elif sign_in_method == "username_password":
+            tableau_auth = TSC.TableauAuth(username, password, site_id=site_id)
+
+        server = TSC.Server(
+            server_url,
+            use_server_version=True,
+        )
+        connection = server.auth.sign_in(tableau_auth)
+        logger.info("Successfully authenticated with Tableau.")
+        return server, connection
+    except Exception as e:
+        logger.error("Failed to connect to Tableau.")
+        if sign_in_method == "username_password":
+            raise ExitCodeException(
+                "Invalid username or password. Please check for typos and try again.",
+                DataVisualization.EXIT_CODE_INVALID_CREDENTIALS,
+            ) from e
+
+        if sign_in_method == "access_token":
+            raise ExitCodeException(
+                "Invalid token name or access token. Please check for typos and try again.",
+                DataVisualization.EXIT_CODE_INVALID_CREDENTIALS,
+            ) from e
+
+
+def get_project_id(server, project_name):
+    """
+    Looks up and returns the project_id of the project_name that was specified.
```

Comment:
Same comment for the `get_**_id` functions - shouldn't those be wrapped in a `try` block? Is there any chance of the following failing for instance?
``` python   
req_option = TSC.RequestOptions()
    req_option.filter.add(
        TSC.Filter(
            TSC.RequestOptions.Field.Name,
            TSC.RequestOptions.Operator.Equals,
            project_name,
        )
    )

    project_matches = server.projects.get(req_options=req_option)
    ```